src/test/operators/join_test.hpp

```diff
@@ -116,6 +117,59 @@ class JoinTest : public BaseTest {
     const auto actual_result = join->get_output();
     EXPECT_TABLE_EQ_UNORDERED(actual_result, expected_result);
 
+    /**
```

Comment:
Also, do you test the secondary join predicates anywhere? I would expect there to be quite a couple of combinations that need to be tested. As we have seen above, NULL handling is entirely uncovered.